### DISCUSSION TITLE
Create branch with same upstream

### DISCUSSION THREAD

**Discussion by bersace** [magit/magit#5243] [Ideas]
Salut,

:fr:  Merci pour magit, c'est génial. Je suis accro depuis des années !

Usually, one creates successive branches against the same upstream (e.g. origin/master). Currently, typing `b c` defaults starting point to `HEAD`. This lead to a branch without upstream. I always retype `origin/master`.

I suggest that default starting point is *current branch upstream* if available.

Is there an option or a hack to do that ?

Regards,
Étienne

**Comment by tarsius** [comment]
Indeed, this could use some tweaking. I never noticed, because I use `vertico-mode` and the starting-point I want to use (`main`, not `origin/main`) is always just one or two `M-p` away (while also being shown, so I know up front how many `M-p` it's going to take).

Anyway, I'll improve this. Might take a few weeks until I get to that.

**Reply by tarsius** [reply]
There should be two new commands.

- `magit-branch-duplicate` creates a branch that is identical to the current (or chosen) branch. Same `HEAD`, same upstream etc., except a different name.
- `magit-branch-respawn` creates a branch with the same upstream, which also points at the same commit as the upstream. This is similar to how the player spawns again at the last "checkpoint" (here the upstream) after dying in some computer games.

**Reply by tarsius** [reply]
These new commands are still on the table (and I might get to that soon), but that is not what this topic was originally about. *That* has been addressed, so I am closing.

**Reply by tarsius** [reply]
I've improved the order in 072ff065510021773059af8cb8387850664d5af0.
